lnd_test.go

```diff
@@ -7567,6 +7567,353 @@ func testRevokedCloseRetributionRemoteHodl(net *lntest.NetworkHarness,
 	assertNodeNumChannels(t, dave, 0)
 }
 
+// testRevokedCloseRetributionAltruistWatchtower tests that Dave is able
+// carry out retribution in the event that she fails in state where the remote
+// commitment output has zero-value.
+func testRevokedCloseRetributionAltruistWatchtower(net *lntest.NetworkHarness,
```

Comment:
This test has a lot of overlap with the existing revocation related itests (step, db state copy, breach). I think much of this can be consolidated, similar to the `assertDLPExecuted` method I added when the SCB DLP tests were introduced.

Reply:
Reading it a bit more, there's a bit less overlap than I originally suspected.

Reply:
Could still refactor this slightly to not cause a large change when adding tests for private towers with rewards.